### PR TITLE
Add afk marks

### PR DESCRIPTION
**Pull request by ardadem**
This is final state  (and most minimalist to not create mess) of my previous pull request. I didn't reopen it because it's contains changed unnecessary commits.

If someone will become afk, mark will be spawn next to tee's name.
2019-07-13-13:59:20-screenshot

### CODE REVIEW

**Comment by bojidar-bg** [comment]
Would this tag dummies as AFK on dummy maps? I think the AFK mark might be distracting for the player.

**Comment by ardadem** [comment]
It's optional, u can set cl_afk_mark to 0.

**Comment by fokkonaut** [comment]
I liked the colored names version more.

**Comment by ardadem** [comment]
Well we have `cl_afk_mark_color` option too, you can use more highlighted color. Or we can set different mark for this.

**Comment by fokkonaut** [comment]
I think this should be a flag, not a NetBool.

**Comment by Learath2** [comment]
Yeah a flag would be more appropriate.

**Comment by heinrich5991** [comment]
This only seems to work if you have enabled nameplates, not if you only show them near the cursor or not at all.

**Comment by ardadem** [comment]
Yes, it's works only if you have enabled nameplates.

File: datasrc/network.py [hunk 1]
```diff
@@ -229,6 +229,7 @@
 	NetObjectEx("DDNetPlayer", "player@netobj.ddnet.tw", [
 		NetIntAny("m_Flags"),
 		NetIntRange("m_AuthLevel", "AUTHED_NO", "AUTHED_ADMIN"),
+                NetBool("m_Afk"),
```
Comment (Learath2) [inline datasrc/network.py]: Move this inside `m_Flags`

File: src/game/client/components/nameplates.cpp [hunk 2]
```diff
@@ -109,6 +111,37 @@ void CNamePlates::RenderNameplate(
 			}
 		}
 
+		if(g_Config.m_ClAfkMark && m_pClient->m_aClients[ClientID].m_Afk)
+		{
+			const char *afkMark = "!";
```
Comment (Learath2) [inline src/game/client/components/nameplates.cpp]:
```suggestion
			const char *AfkMark = "!";
```
We don't do camelCase

File: src/game/client/components/nameplates.cpp [hunk 3]
```diff
@@ -109,6 +111,37 @@ void CNamePlates::RenderNameplate(
...
+				MapscreenToGroup(m_pClient->m_pCamera->m_Center.x, m_pClient->m_pCamera->m_Center.y, Layers()->GameGroup());
+
+				m_aNamePlates[ClientID].m_AfkMarkWidth = TextRender()->TextWidth(0, FontSize - FontOffsetAfkMark, afkMark, -1);
+
+				m_aNamePlates[ClientID].m_AfkMarkContainerIndex = TextRender()->CreateTextContainer(&Cursor, afkMark);
+				Graphics()->MapScreen(ScreenX0, ScreenY0, ScreenX1, ScreenY1);
+			}
+		}
+
+		ColorRGBA rgbAfkM = color_cast<ColorRGBA>(ColorHSLA(g_Config.m_ClAfkMarkColor));
```
Comment (Learath2) [inline src/game/client/components/nameplates.cpp]:
```suggestion
		ColorRGBA RgbAfkM = color_cast<ColorRGBA>(ColorHSLA(g_Config.m_ClAfkMarkColor));
```
Proper casing and maybe a better name for it.

**Review by Learath2 — CHANGES_REQUESTED** [review]
A couple small nits, after that it looks ready to merge for me.

**Comment by ardadem** [comment]
What about now?

**Comment by def-** [comment]
bors r+
Thanks!

**Comment by ardadem** [comment]
@def- This pull request doesn't include paused ones. You removed eye blink function but now we can't see who is paused or playing.

**Comment by Learath2** [comment]
Also create a new flags array for the extended flags, as like this you are
wasting flagspace in DDNetPlayer.

bors r-

**Comment by 12pm** [comment]
> @def- This pull request doesn't include paused ones. You removed eye blink function but now we can't see who is paused or playing.

Why doesn’t it include paused tees? It’s effectively the same as being afk (you’re unable to react)

**Comment by ardadem** [comment]
It would be messy otherwise

**Comment by 12pm** [comment]
But what's the purpose of this feature then if it doesn't fix the "bug" of being able to set afk eyes yourself?

**Pull request merged**